code/game/machinery/electrolyzer.dm

```diff
@@ -4,15 +4,14 @@
 /obj/machinery/electrolyzer
 	anchored = FALSE
 	density = TRUE
-	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN | INTERACT_MACHINE_OFFLINE
 	icon = 'icons/obj/atmos.dmi'
 	icon_state = "electrolyzer-off"
 	name = "space electrolyzer"
 	desc = "Thanks to the fast and dynamic response of our electrolyzers, on-site hydrogen production is guaranteed. Warranty void if used by clowns"
 	max_integrity = 250
 	circuit = /obj/item/circuitboard/machine/electrolyzer
-	/// We don't use area power, we always use the cell
-	use_power = NO_POWER_USE
+	use_power = ACTIVE_POWER_USE
```

Comment:
This should only be ACTIVE_POWER_USE when on = TRUE
In addition you need to actually set a power draw

Reply:
i did

Reply:
It only draw power when the machine is processing

Reply:
i put ACTIVE_POWER_USE there for it to use apc power

Reply:
I put use_power = ACTIVE_POWER_USE there for it to use active_power_usage, when its processing active_power_usage will be changed. When it's not active_power_usage will be 0

Reply:
also active_power_usage will be based on the efficiency how much gas near by